Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -290,6 +293,27 @@ public static void DeleteDirectoryContentsSafe(string path, bool ignoreErrors =
             DeleteDirectoryContentsSafe(Instance.DirectoryInfo.FromDirectoryName(path), ignoreErrors);
         }
 
+        public static bool IsFileSystemReadOnly()
+        {
+            if (TmpFolder.StartsWith("%WEBROOT_PATH%", StringComparison.OrdinalIgnoreCase))
+            {
+                // not able to check, return false since by default we are expecting none readonly file system
+                return false;
+            }
+
+            try
+            {
+                string folder = Path.Combine(TmpFolder, Guid.NewGuid().ToString());
+                CreateDirectory(folder);
+                DeleteDirectorySafe(folder, ignoreErrors: false);
+                return false;
+            }
```

Comment:
Can this value be cached? How often might this change, or can we safely assume that after we check once, for the lifetime of the app domain it won't change?